### ISSUE TITLE
Intermittent OpenJDK 11.0.8 (and 11.0.12) Crash in GC

### ISSUE THREAD

**Issue by kolehey** [adoptium/adoptium-support#344]
## Summary

During the JVM runtime excessive of 24 hours JVM intermittently crashes on G1GC Thread.
The below logs are from A1 node

    # A fatal error has been detected by the Java Runtime Environment:
    #
    #  SIGSEGV (0xb) at pc=0x00007f183fe209a0, pid=25538, tid=25582
    #
    # JRE version: OpenJDK Runtime Environment AdoptOpenJDK (11.0.8+10) (build 11.0.8+10)
    # Java VM: OpenJDK 64-Bit Server VM AdoptOpenJDK (11.0.8+10, mixed mode, tiered, g1 gc, linux-amd64)
    # Problematic frame:
    # V  [libjvm.so+0xf419a0]
    #
    # Core dump will be written. Default location: Core dumps may be processed with "/opt/dynatrace/oneagent/agent/rdp -p %p -P %P -e %e -s %s" (or dumping to /opt/PingIdentity/PingDirectory/bin/core.25538)
    #
    # If you would like to submit a bug report, please visit:
    #   https://github.com/AdoptOpenJDK/openjdk-support/issues
    #

The crash occurs on the below running thread:

    =>0x00007f1838a2f800 GCTaskThread "GC Thread#0" [stack: 0x00007f18176ff000,0x00007f18177ff000] [id=25582]

    *Filename: B1_hs_err_pid14816.log_gccrash

A similar problem occurs on B1 node as well

    # A fatal error has been detected by the Java Runtime Environment:
    #
    #  SIGSEGV (0xb) at pc=0x00007f9742cc04a5, pid=12994, tid=13127
    #
    # JRE version: OpenJDK Runtime Environment AdoptOpenJDK (11.0.8+10) (build 11.0.8+10)
    # Java VM: OpenJDK 64-Bit Server VM AdoptOpenJDK (11.0.8+10, mixed mode, tiered, g1 gc, linux-amd64)
    # Problematic frame:
    # V  [libjvm.so+0x7ae4a5]  G1ParScanThreadState::copy_to_survivor_space(InCSetState, oopDesc*, markOopDesc*)+0x35
    #
    # Core dump will be written. Default location: Core dumps may be processed with "/opt/dynatrace/oneagent/agent/rdp -p %p -P %P -e %e -s %s" (or dumping to /opt/PingIdentity/PingDirectory/bin/core.12994)
    #
    # If you would like to submit a bug report, please visit:
    #   https://github.com/AdoptOpenJDK/openjdk-support/issues
    #

Crash thread at:

    =>0x00007f8e9400a800 GCTaskThread "GC Thread#1" [stack: 0x00007f8e471b5000,0x00007f8e472b5000] [id=13127]

## Initial Suspicion/Observation
Initial Suspicion/Observations

    - Since the crash occurred on two different machines and different JVM uptime it points to a possible thread contention on a GC Thread where the heap is getting corrupted for an unknown reason.
    - The customer has removed third-party code like DynaTrace OneAgent and restart the JVM, however, they experience similar issues over time. G1GC thread fails.
    - Since this is an Amazon EC to instance, they have allocated 64GB of physical RAM to the server and JVM is sized with 32GB heap. However, since G1GC is being used -XX:NewSize=1g and -XX:MaxNewSize=4g. As part of PD java. properties when we change JVM to use start-server.gc-g1-args we still see NewSize and MaxNewSize. Is this a mistake in PD?
    As per Oracle, they recommend if we use G1GC young gen allocation size should not be specified ie:https://docs.oracle.com/javase/9/gctuning/garbage-first-garbage-collector.htm#JSGCT-GUID-98E80C82-24D8-41D4-BC39-B2583F04F1FF

If we take a look at the Java crash dump we see the following: in the global flasg

    [Global flags]
         bool AlwaysTenure                             = false                                     {product} {command line}
         intx CICompilerCount                          = 12                                        {product} {ergonomic}
         uint ConcGCThreads                            = 1                                         {product} {command line}
         uint G1ConcRefinementThreads                  = 13                                        {product} {ergonomic}
       size_t G1HeapRegionSize                         = 16777216                                  {product} {ergonomic}
        uintx GCDrainStackTargetSize                   = 64                                        {product} {ergonomic}
         bool HeapDumpOnOutOfMemoryError               = true                                   {manageable} {command line}
       size_t InitialHeapSize                          = 34359738368                               {product} {command line}
        uintx InitialTenuringThreshold                 = 2                                         {product} {ergonomic}
        uintx InitiatingHeapOccupancyPercent           = 80                                        {product} {command line}
       size_t MarkStackSize                            = 4194304                                   {product} {ergonomic}
       size_t MaxHeapSize                              = 34359738368                               {product} {command line}
       size_t MaxNewSize                               = 4294967296                                {product} {command line}
        uintx MaxTenuringThreshold                     = 2                                         {product} {command line}
       size_t MinHeapDeltaBytes                        = 16777216                                  {product} {ergonomic}
         bool NeverTenure                              = false                                     {product} {command line}
       size_t NewSize                                  = 2147483648                                {product} {command line}
        uintx NonNMethodCodeHeapSize                   = 7594288                                {pd product} {ergonomic}
        uintx NonProfiledCodeHeapSize                  = 122031976                              {pd product} {ergonomic}
         bool OmitStackTraceInFastThrow                = false                                     {product} {command line}
         bool PrintCommandLineFlags                    = true                                      {product} {command line}
        uintx ProfiledCodeHeapSize                     = 122031976                              {pd product} {ergonomic}
         intx RefDiscoveryPolicy                       = 1                                         {product} {command line}
        uintx ReservedCodeCacheSize                    = 251658240                              {pd product} {ergonomic}
         bool SegmentedCodeCache                       = true                                      {product} {ergonomic}
         bool UseBiasedLocking                         = true                                      {product} {command line}
         bool UseG1GC                                  = true          

More recently they are also getting crash on PD run thread:
File: A1_hs_err_pid29686.log_gccrash

    #  SIGSEGV (0xb) at pc=0x00007f7044dee4d6, pid=29686, tid=14980
    #
    # JRE version: OpenJDK Runtime Environment AdoptOpenJDK (11.0.8+10) (build 11.0.8+10)
    # Java VM: OpenJDK 64-Bit Server VM AdoptOpenJDK (11.0.8+10, mixed mode, tiered, g1 gc, linux-amd64)
    # Problematic frame:
    # J 9459 c2 java.util.HashSet.add(Ljava/lang/Object;)Z java.base@11.0.8 (20 bytes) @ 0x00007f7044dee4d6 [0x00007f7044dee480+0x0000000000000056]

Failed on:

    =>0x00007f67600d2800 JavaThread "Server monitor info collector using connection LDAPConnection(connected to 10.128.87.219:1636)  (8.2.0.2-20210204225928.000Z-4d7c823d)" daemon [_thread_in_Java, id=14980, stack(0x00007f65f4a97000,0x00007f65f4b98000)]

running Stack

    Current thread (0x00007f67600d2800):  JavaThread "Server monitor info collector using connection LDAPConnection(connected to 10.128.87.219:1636)  (8.2.0.2-20210204225928.000Z-4d7c823d)" daemon [_thread_in_Java, id=14980, stack(0x00007f65f4a97000,0x00007f65f4b98000)]

    Stack: [0x00007f65f4a97000,0x00007f65f4b98000],  sp=0x00007f65f4b96890,  free space=1022k
    Native frames: (J=compiled Java code, A=aot compiled Java code, j=interpreted, Vv=VM code, C=native code)
    J 9459 c2 java.util.HashSet.add(Ljava/lang/Object;)Z java.base@11.0.8 (20 bytes) @ 0x00007f7044dee4d6 [0x00007f7044dee480+0x0000000000000056]
    J 9881 c2 java.util.AbstractCollection.addAll(Ljava/util/Collection;)Z java.base@11.0.8 (42 bytes) @ 0x00007f7044ed6b18 [0x00007f7044ed6620+0x00000000000004f8]
    J 30737 c2 com.unboundid.directory.server.api.DirectoryThread.run()V (1552 bytes) @ 0x00007f704614f930 [0x00007f704614df40+0x00000000000019f0]
    v  ~StubRoutines::call_stub
    V  [libjvm.so+0x88d1c9]  JavaCalls::call_helper(JavaValue*, methodHandle const&, JavaCallArguments*, Thread*)+0x3b9
    V  [libjvm.so+0x88b16d]  JavaCalls::call_virtual(JavaValue*, Handle, Klass*, Symbol*, Symbol*, Thread*)+0x1ed
    V  [libjvm.so+0x936fec]  thread_entry(JavaThread*, Thread*)+0x6c
    V  [libjvm.so+0xdcc2ea]  JavaThread::thread_main_inner()+0x1ba
    V  [libjvm.so+0xdc8d7f]  Thread::call_run()+0x14f
    V  [libjvm.so+0xc175ae]  thread_native_entry(Thread*)+0xee

Is this also related to JVM heap issue?
V [libjvm.so+0xc175ae] thread_native_entry(Thread*)+0xee

See Ref:
https://bugs.openjdk.java.net/browse/JDK-8246219
https://bugs.openjdk.java.net/browse/JDK-8170451

Actual crash dump has been turned off at ulimit (core dump). If the customer enables this it will produce very large dumps (closer to perhaps 32GB), do you want these attached to the case? 

## Additional Crash
File: hs_err_pid24779.log
Version: 11.0.11+9

    # A fatal error has been detected by the Java Runtime Environment:
    #
    #  SIGSEGV (0xb) at pc=0x00007f0c3afac118, pid=24779, tid=24873
    #
    # JRE version: OpenJDK Runtime Environment Corretto-11.0.11.9.1 (11.0.11+9) (build 11.0.11+9-LTS)
    # Java VM: OpenJDK 64-Bit Server VM Corretto-11.0.11.9.1 (11.0.11+9-LTS, mixed mode, tiered, g1 gc, linux-amd64)
    # Problematic frame:
    # V  [libjvm.so+0x7a0118]  G1FullGCMarker::complete_marking(GenericTaskQueueSet<OverflowTaskQueue<oopDesc*, (MemoryType)5, 131072u>, (MemoryType)5>*, GenericTaskQueueSet<OverflowTaskQueue<ObjArrayTask, (MemoryType)5, 131072u>, (MemoryType)5>*, ParallelTaskTerminator*)+0x1498

Running Stack

    Current thread (0x00007f03b8005800):  GCTaskThread "GC Thread#4" [stack: 0x00007f0367cfc000,0x00007f0367dfc000] [id=24873]
    
    Stack: [0x00007f0367cfc000,0x00007f0367dfc000],  sp=0x00007f0367dfac80,  free space=1019k
    Native frames: (J=compiled Java code, A=aot compiled Java code, j=interpreted, Vv=VM code, C=native code)
    V  [libjvm.so+0x7a0118]  G1FullGCMarker::complete_marking(GenericTaskQueueSet<OverflowTaskQueue<oopDesc*, (MemoryType)5, 131072u>, (MemoryType)5>*, GenericTaskQueueSet<OverflowTaskQueue<ObjArrayTask, (MemoryType)5, 131072u>, (MemoryType)5>*, ParallelTaskTerminator*)+0x1498
    V  [libjvm.so+0x79e675]  G1FullGCMarkTask::work(unsigned int)+0xe5
    V  [libjvm.so+0xeed723]  GangWorker::loop()+0x43
    V  [libjvm.so+0xe55e2d]  Thread::call_run()+0x14d
    V  [libjvm.so+0xc09a6c]  thread_native_entry(Thread*)+0xec

## Workarounds

    Update Java to latest OpenJDK build or use Oracle JDK latest
    Change G1GC to a stop the world CMS method to rule of if threading issue
    Reduce JVM heap based on verbose GC log analysis.

## Triaging info

Java version: AdoptOpenJDK (11.0.8+10) (build 11.0.8+10)

What is your operating system and platform?

    Amazon Linux release 2 (Karoo)
    16 CPU cores (Intel(R) Xeon(R) Platinum 8175M CPU @ 2.50GHz)
    62 GB Physical RAM

[STAGING-13210.zip](https://github.com/adoptium/adoptium-support/files/6984918/STAGING-13210.zip)

**Comment by kolehey** [comment]
If needed, we have an additional core dump file. Holding off on uploading it due to filesize (>38 GB), unless requested.

**Comment by karianna** [comment]
@kolehey You'll need to upgrade to 11.0.12

**Comment by kolehey** [comment]
@karianna Is this a known issue at the moment? More specifically, has this issue been addressed and resolved in 11.0.12?

**Comment by karianna** [comment]
I believe so.  You can check the release notes at adoptium.net/release_notes.html and adoptopenjdk.net/release_notes.html

**Comment by kqarryzada** [comment]
@karianna, thanks for the guidance. We advised our client to upgrade to 11.0.12, and a few days later, they had another SIGSEGV, though it appears different:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fef59f69fc9, pid=16304, tid=6861
#
# JRE version: OpenJDK Runtime Environment Corretto-11.0.12.7.1 (11.0.12+7) (build 11.0.12+7-LTS)
# Java VM: OpenJDK 64-Bit Server VM Corretto-11.0.12.7.1 (11.0.12+7-LTS, mixed mode, tiered, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0xceefc9]  SharedRuntime::generate_class_cast_message(Klass*, Klass*, Symbol*)+0x59
#
# Core dump will be written. Default location: /opt/PingIdentity/PingDirectory/bin/core.16304
#
# If you would like to submit a bug report, please visit:
#   https://github.com/corretto/corretto-11/issues/
#
```

I believe this would still be evaluated by Adoptium, but please correct me if I'm wrong about that. My question is whether we should provide more information here, or if this is different enough to require a new ticket. Please let me know.

**Comment by karianna** [comment]
You're using an Amazon Corretto build - you should report this over at https://github.com/corretto/corretto-11

**Comment by devendrajoshi** [comment]
We have been facing the same issue for past few weeks. In our case, JVM runtime is only couple of hours. Here is the error:

#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fc04c7467e4, pid=14, tid=35
#
# JRE version: OpenJDK Runtime Environment AdoptOpenJDK (11.0.8+10) (build 11.0.8+10)
# Java VM: OpenJDK 64-Bit Server VM AdoptOpenJDK (11.0.8+10, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x7ae7e4]  G1ParScanThreadState::copy_to_survivor_space(InCSetState, oopDesc*, markOopDesc*)+0x374
#
# Core dump will be written. Default location: //core.14
#
# If you would like to submit a bug report, please visit:
#   https://github.com/AdoptOpenJDK/openjdk-support/issues
#


We are seeing this error in roughly 1 out of 20 runs. Can someone please confirm if this issue is fixed in 11.0.12? Please let me know if more information is needed.

**Comment by karianna** [comment]
> We have been facing the same issue for past few weeks. In our case, JVM runtime is only couple of hours. Here is the error:
> 
> ![image](https://user-images.githubusercontent.com/26906284/131941167-0718840a-c6e4-4eea-acb8-4949af911e57.png)
> 
> We are seeing this error in roughly 1 out of 20 runs. Can someone please confirm if this issue is fixed in 11.0.12? Please let me know if more information is needed.

Please use the latest version 11.0.12+7 and report back here if it's still an issue (this crash report is different from the other poster's).

**Comment by devendrajoshi** [comment]
> > We have been facing the same issue for past few weeks. In our case, JVM runtime is only couple of hours. Here is the error:
> > ![image](https://user-images.githubusercontent.com/26906284/131941167-0718840a-c6e4-4eea-acb8-4949af911e57.png)
> > We are seeing this error in roughly 1 out of 20 runs. Can someone please confirm if this issue is fixed in 11.0.12? Please let me know if more information is needed.
> 
> Please use the latest version 11.0.12+7 and report back here if it's still an issue (this crash report is different from the other poster's).

Hello Karianna,
We tried using version 11.0.12+7, but are facing the same issue:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fd2afe2ad04, pid=14, tid=34
#
# JRE version: OpenJDK Runtime Environment Temurin-11.0.12+7 (11.0.12+7) (build 11.0.12+7)
# Java VM: OpenJDK 64-Bit Server VM Temurin-11.0.12+7 (11.0.12+7, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x7f4d04]  G1ParScanThreadState::copy_to_survivor_space(InCSetState, oopDesc*, markOopDesc*)+0x374
#
# Core dump will be written. Default location: //core.14
#
# If you would like to submit a bug report, please visit:
#   https://github.com/adoptium/adoptium-support/issues
#

```

Any other suggestions?

**Comment by gabihodoroaga** [comment]
The JVM crashes regularly on GCP VM.
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f1b337a470d, pid=96328, tid=96349
#
# JRE version: OpenJDK Runtime Environment Temurin-11.0.13+8 (11.0.13+8) (build 11.0.13+8)
# Java VM: OpenJDK 64-Bit Server VM Temurin-11.0.13+8 (11.0.13+8, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x7c270d]  G1ParScanThreadState::copy_to_survivor_space(InCSetState, oopDesc*, markOopDesc*)+0x36d
#
# Core dump will be written. Default location: Core dumps may be processed with "/sbin/crash_reporter --user=%P:%s:%u:%g:%f" (or dumping to /home/project/test_project/dataflow/ble_to_bigquery/kotlin/app/core.96328)
#
# If you would like to submit a bug report, please visit:
#   https://github.com/adoptium/adoptium-support/issues
#

---------------  S U M M A R Y ------------

Command Line: -Dorg.gradle.internal.worker.tmpdir=/home/project/test_project/dataflow/ble_to_bigquery/kotlin/app/build/tmp/test/work -Dorg.gradle.native=false --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED -Xmx512m -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -ea worker.org.gradle.process.internal.worker.GradleWorkerMain 'Gradle Test Executor 1'

Host: Intel(R) Xeon(R) CPU @ 3.10GHz, 4 cores, 15G, Alpine Linux v3.14
Time: Thu Jan 20 11:52:30 2022 UTC elapsed time: 19.243935 seconds (0d 0h 0m 19s)

**Comment by karianna** [comment]
> The JVM crashes regularly on GCP VM.
> 
> ```
> #
> # A fatal error has been detected by the Java Runtime Environment:
> #
> #  SIGSEGV (0xb) at pc=0x00007f1b337a470d, pid=96328, tid=96349
> #
> # JRE version: OpenJDK Runtime Environment Temurin-11.0.13+8 (11.0.13+8) (build 11.0.13+8)
> # Java VM: OpenJDK 64-Bit Server VM Temurin-11.0.13+8 (11.0.13+8, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
> # Problematic frame:
> # V  [libjvm.so+0x7c270d]  G1ParScanThreadState::copy_to_survivor_space(InCSetState, oopDesc*, markOopDesc*)+0x36d
> #
> # Core dump will be written. Default location: Core dumps may be processed with "/sbin/crash_reporter --user=%P:%s:%u:%g:%f" (or dumping to /home/project/test_project/dataflow/ble_to_bigquery/kotlin/app/core.96328)
> #
> # If you would like to submit a bug report, please visit:
> #   https://github.com/adoptium/adoptium-support/issues
> #
> 
> ---------------  S U M M A R Y ------------
> 
> Command Line: -Dorg.gradle.internal.worker.tmpdir=/home/project/test_project/dataflow/ble_to_bigquery/kotlin/app/build/tmp/test/work -Dorg.gradle.native=false --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED -Xmx512m -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -ea worker.org.gradle.process.internal.worker.GradleWorkerMain 'Gradle Test Executor 1'
> 
> Host: Intel(R) Xeon(R) CPU @ 3.10GHz, 4 cores, 15G, Alpine Linux v3.14
> Time: Thu Jan 20 11:52:30 2022 UTC elapsed time: 19.243935 seconds (0d 0h 0m 19s)
> ```

We are releasing 11.0.14 over the coming 4-5 working days.  I'm recommend trying that out once it lands (a few GC crashes were fixed).

**Comment by huyang327** [comment]
Crashing after 12 hours
![WechatIMG4](https://user-images.githubusercontent.com/79496601/161362819-506c191e-5f1a-4d90-bdf0-50dcae2796fe.png)
![WechatIMG5](https://user-images.githubusercontent.com/79496601/161362838-ceadfcbe-5b7b-49ef-84c9-6de44ff466e9.png)

**Comment by karianna** [comment]
@huyang327 - Can you try with 11.0.15 - a bunch of GC fixes went in

**Comment by huyang327** [comment]
> @huyang327 - Can you try with 11.0.15 - a bunch of GC fixes went in
I checked the problem. I checked the memory of my machine with memtest and found that there was a problem with the memory. We replaced the server and the problem did not appear.

**Comment by Rudrapuneet** [comment]
We are still facing this issue 

Current Version 👍 openjdk version "11.0.15" 2022-04-19 LTS
OpenJDK Runtime Environment 18.9 (build 11.0.15+9-LTS)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.15+9-LTS, mixed mode, sharing)

# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f0a49052344, pid=22443, tid=22445
#
# JRE version: OpenJDK Runtime Environment 18.9 (11.0.15+9) (build 11.0.15+9-LTS)
# Java VM: OpenJDK 64-Bit Server VM 18.9 (11.0.15+9-LTS, mixed mode, sharing, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x80a344]  G1ParScanThreadState::copy_to_survivor_space(InCSetState, oopDesc*, markOopDesc*)+0x3f4
#
# Core dump will be written. Default location: Core dumps may be processed with "/bin/false" (or dumping to /opt/redwood/j2ee/cluster/server1/logs/core.22443)
#
# If you would like to submit a bug report, please visit:
#   https://bugzilla.redhat.com/enter_bug.cgi?product=Red%20Hat%20Enterprise%20Linux%207&component=java-11-openjdk
#
Current thread (0x00007f0a44030000):  GCTaskThread "GC Thread#0" [stack: 0x00007f0a2b8ef000,0x00007f0a2b9ef000] [id=22445]

**Comment by karianna** [comment]
@Rudrapuneet that's for the Red Hat Build of OpenJDK (you'll want to contact them about it as suggested in the link at the end of the error message).  11.0.16 is also now out so you may want to try that.

**Comment by douglasawh** [comment]
I think this may be a generic JVM bug, not one specific to Adoptium. We are seeing it in 11.0.18 (though not Adoptium). Was just reading through to see if there was a workaround suggested.

**Comment by enicolas31** [comment]
@karianna Using version 11.0.19+7 we still have the JVM crash :
```
# JRE version: OpenJDK Runtime Environment Temurin-11.0.19+7 (11.0.19+7) (build 11.0.19+7)
# Java VM: OpenJDK 64-Bit Server VM Temurin-11.0.19+7 (11.0.19+7, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x661b7c]  oopDesc::size_given_klass(Klass*)+0x2c
```
Thanks for you feedback

**Comment by karianna** [comment]
https://bugs.openjdk.org/browse/JDK-8303239

**Comment by karianna** [comment]
Hi folks, curious to hear if 11.0.20.1 has resolved the issue for folks.

**Comment by beamerblvd** [comment]
For what it's worth, I'll chime in on the problems that I've had the last few days working with GitHub Workflows using Adoptium. This may or may not be the same problem, but it sure looks similar:

- This crash has occurred for me on: macOS 11, macOS 13, Ubuntu 20.04, and Ubuntu 22.04
- It has occurred running Java: 11.0.21, 17.0.9, and 21.0.1 (intermittently on all the platforms above)
- It does not require the JVM to run for a long time, but rather it crashes in just milliseconds (and it rarely crashes, but when it does, it's fast)
- The problematic frames I've seen so far are listed below, but they are always in one of the "GC Thread#n" threads:

```
# Problematic frame:
# V  [libjvm.dylib+0x4f4e0e]  G1ParScanThreadState::copy_to_survivor_space(G1HeapRegionAttr, oopDesc*, markWord)+0x3e

# Problematic frame:
# V  [libjvm.so+0x75e0ef]  G1ParCopyClosure<(G1Barrier)0, false>::do_oop(oopDesc**)+0x3f
```

I can provide a few entire `hs_err*.log` files for the above if needed. I can also try to provide other troubleshooting info if needed, but keep in mind that it's not trivial to get information like the above from GitHub Workflows (I got what I got by adding ` || (ls -al && cat hs_err*.log && false)` to the end of the `mvn` command).

I'm also going to install Adoptium on my local macOS and Ubuntu machines and try to replicate the problem by running the `mvn` command in a loop, where hopefully I can get more information. It's possible that I'll determine the problem is unrelated to this (or be unable to replicate it outside of GitHub Workflows), but I wanted to go ahead and put this out there in case I get any advice on tracking down the problem and/or this information helps you in some way.

**Comment by karianna** [comment]
@beamerblvd hs_err files would be most helpful thanks.  Is there are particular type of workload you are running?

**Comment by shark-mk** [comment]
[hs_err_pid173.log](https://github.com/user-attachments/files/16261450/hs_err_pid173.log)
Please take a look at my error message.
Running in a virtual machine installed in Hyper-V on Windows 11.
@karianna

**Comment by huyang327** [comment]
> [hs_err_pid173.log](https://github.com/user-attachments/files/16261450/hs_err_pid173.log) Please take a look at my error message. Running in a virtual machine installed in Hyper-V on Windows 11. @karianna

The method that caused the error is copy_to_survivor_space, and the address of si_addr is 0x00000007c1b816d8, which seems to be within the range of Compressed class space size: 1073741824 Address: 0x00000007c0000000.
It looks a bit strange, why is the copy_to_survivor_space method related to the Compressed class space memory?

**Comment by karianna** [comment]
> > [hs_err_pid173.log](https://github.com/user-attachments/files/16261450/hs_err_pid173.log) Please take a look at my error message. Running in a virtual machine installed in Hyper-V on Windows 11. @karianna
> 
> The method that caused the error is copy_to_survivor_space, and the address of si_addr is 0x00000007c1b816d8, which seems to be within the range of Compressed class space size: 1073741824 Address: 0x00000007c0000000. It looks a bit strange, why is the copy_to_survivor_space method related to the Compressed class space memory?

This seems to be from a Java 1.8.0_412 instance that's from Oracle, you'll want to report that to them.

**Comment by karianna** [comment]
The latest quarterly release is out so folks can try 1.8.0_422, 11.0.24, 17.0.12 or 21.0.4

**Comment by shark-mk** [comment]
> The latest quarterly release is out so folks can try 1.8.0_422, 11.0.24, 17.0.12 or 21.0.4

Thank you for your reply, I will update the version and try it out.

**Comment by shark-mk** [comment]
@karianna 
1.8.0_422
[hs_err_pid9757.log](https://github.com/user-attachments/files/16438083/hs_err_pid9757.log)

**Comment by huyang327** [comment]
> @karianna 1.8.0_422 [hs_err_pid9757.log](https://github.com/user-attachments/files/16438083/hs_err_pid9757.log)

According to your log:
par new generation memory is in the range of 0x00000000aa000000, 0x00000000ca000000
concurrent mark-sweep generation memory is in the range of
0x0000000455000000, 0x00000005a0016000
from space memory is in the range of
0x00000000c6cd0000, 0x00000000c99572e0
to space memory is in the range of
0x00000000c39a0000, 0x00000000c5aea070
Why does the program try to access memory when it crashes
si_addr: 0x00000004500ad580
This memory address is not in any memory area where gc works.
Have you used memtest to check if there is any problem with the server memory?
